src/modules/roc_audio/resampler_reader.cpp

```diff
@@ -18,10 +18,13 @@ namespace audio {
 ResamplerReader::ResamplerReader(IReader& reader,
                                  IResampler& resampler,
                                  core::BufferPool<sample_t>& buffer_pool,
-                                 size_t frame_size)
+                                 core::nanoseconds_t frame_length,
+                                 size_t sample_rate,
+                                 roc::packet::channel_mask_t ch_mask)
     : resampler_(resampler)
     , reader_(reader)
-    , frame_size_(frame_size)
+    , frame_size_(packet::ns_to_size(frame_length, sample_rate,
+                  ch_mask))
```

Comment:
We need to check if frame_size_ is zero, as we do it in ResamplerWriter.

Reply:
done.